packages/web3-eth-contract/src/index.js

```diff
@@ -877,131 +875,134 @@ Contract.prototype._executeMethod = function _executeMethod(){
-                    name: 'sendTransaction',
-                    call: 'eth_sendTransaction',
-                    params: 1,
-                    inputFormatter: [formatters.inputTransactionFormatter],
-                    requestManager: _this._parent._requestManager,
-                    accounts: _this.constructor._ethAccounts || _this._ethAccounts, // is eth.accounts (necessary for wallet signing)
-                    defaultAccount: _this._parent.defaultAccount,
-                    defaultBlock: _this._parent.defaultBlock,
-                    transactionBlockTimeout: _this._parent.transactionBlockTimeout,
-                    transactionConfirmationBlocks: _this._parent.transactionConfirmationBlocks,
-                    transactionPollingTimeout: _this._parent.transactionPollingTimeout,
-                    defaultCommon: _this._parent.defaultCommon,
-                    defaultChain: _this._parent.defaultChain,
-                    defaultHardfork: _this._parent.defaultHardfork,
-                    handleRevert: _this._parent.handleRevert,
-                    extraFormatters: extraFormatters,
-                    abiCoder: abi
-                })).createFunction();
-
-                return sendTransaction(args.options, args.callback);
+                    return receipt;
+                },
+                contractDeployFormatter: function (receipt) {
+                    var newContract = _this._parent.clone();
+                    newContract.options.address = receipt.contractAddress;
+                    return newContract;
+                }
+            };
 
-        }
+            var sendTransaction = (new Method({
+                name: 'sendTransaction',
+                call: 'eth_sendTransaction',
+                params: 1,
+                inputFormatter: [formatters.inputTransactionFormatter],
+                requestManager: _this._parent._requestManager,
+                accounts: _this.constructor._ethAccounts || _this._ethAccounts, // is eth.accounts (necessary for wallet signing)
+                defaultAccount: _this._parent.defaultAccount,
+                defaultBlock: _this._parent.defaultBlock,
+                transactionBlockTimeout: _this._parent.transactionBlockTimeout,
+                transactionConfirmationBlocks: _this._parent.transactionConfirmationBlocks,
+                transactionPollingTimeout: _this._parent.transactionPollingTimeout,
+                defaultCommon: _this._parent.defaultCommon,
+                defaultChain: _this._parent.defaultChain,
+                defaultHardfork: _this._parent.defaultHardfork,
+                handleRevert: _this._parent.handleRevert,
+                extraFormatters: extraFormatters,
+                abiCoder: abi
+            })).createFunction();
+
+            return sendTransaction(args.options, args.callback);
+
+        default:
+            throw new Error('Method "' + args.type + '" not implemented.');
```

Comment:
Could you if you open another PR for this project reference additions like this newly added error in the description of the pull request?